infra/experimental/sanitizers/ExecSan/execSan.cpp

```diff
@@ -266,6 +272,27 @@ void inspect_for_corruption(pid_t pid, const user_regs_struct &regs) {
   match_error_pattern(buffer, g_shell_pids[pid]);
 }
 
+void inspect_for_arbitrary_file_open(pid_t pid, const user_regs_struct &regs) {
+  // Inspect a PID's register for the sign of arbitrary file open.
+  std::string path = read_string(pid, regs.rsi, kRootDirMaxLength);
+  if (!path.length()) {
+    return;
+  }
+  if (path.substr(0, kFzAbsoluteDirectory.length()) == kFzAbsoluteDirectory) {
+    report_bug(kArbitraryFileOpenError);
+  }
+  if (path[0] == '/' && path.length() > 1) {
+    size_t found = path.find('/', 1);
+    if (found != std::string::npos) {
```

Comment:
if `found == std::string::npos`, can we just take the entire string? 

i.e. we should be able to detect both `/blahdgfdf` and `/blahdgfdf/`. Handling the former makes it slightly easier to discover.

Reply:
Done :-)